src/jd_client/job_declarator/mod.rs

```diff
@@ -234,6 +234,19 @@ impl JobDeclarator {
             tokio::task::yield_now().await;
         }
         super::IS_CUSTOM_JOB_SET.store(false, std::sync::atomic::Ordering::Release);
+
+        let tx_list_ = {
```

Comment:
There should be a configurable duration (default of 30secs) to wait for a transactions to arrive. If no transaction arrives within that time, fall back to using the transaction from `TemplateRx`.